### PR TITLE
Add DOCKER_EXPERIMENTAL environment variable

### PR DESCRIPTION
**Pull request by icecrime**
The `DOCKER_EXPERIMENTAL` environment variable drives the activation of the 'experimental' build tag.

Contributes to #13023 but doesn't close it just yet.

### CODE REVIEW

**Comment by jfrazelle** [comment]
LGTM

File: hack/make.sh [hunk 1]
```diff
@@ -93,6 +93,10 @@ if [ ! "$GOPATH" ]; then
 	exit 1
 fi
 
+if [ -z "$DOCKER_EXPERIMENTAL" ]; then
```
Comment (thaJeztah) [inline hack/make.sh]: Should the `-z` be here?

Comment (jfrazelle) [inline hack/make.sh]: wait ya, no -z

Comment (thaJeztah) [inline hack/make.sh]: I know you want all those fancy new features in, but this isn't the way to do that LOL

**Comment by icecrime** [comment]
Updated!

**Comment by thaJeztah** [comment]
Much better! LGTM

**Comment by jfrazelle** [comment]
LGTM

**Comment by jfrazelle** [comment]
will wait for bash boss @tianon :angel:

**Comment by tianon** [comment]
Gah sorry, mobile misclick :/

**Comment by tianon** [comment]
Do you think it would be good to show a warning to people when this is set so they don't get it accidentally?

**Comment by icecrime** [comment]
Updated!

**Comment by jfrazelle** [comment]
I thinnk maybe he meant like when you run the binary, but ya maybe we should put in docker info

**Comment by thaJeztah** [comment]
Both make sense, but +1 on info *and* `docker (--)version`; adding an `-experimental` suffix to the version.

**Comment by icecrime** [comment]
Oh god, *needs some sleep*

**Comment by icecrime** [comment]
Updated!

**Comment by jfrazelle** [comment]
LGTM :)

**Comment by thaJeztah** [comment]
Sorry, I'm probably only adding noise here, but should `hack/make.sh` also be updated, so that the binaries/artifacts are named differently? Not sure if EXPERIMENTAL builds will be part of the automated builds, and if that would cause problems otherwise

**Comment by jfrazelle** [comment]
i think this is fine for now

**Comment by tianon** [comment]
Heh, I did actually mean while compiling, but docker info is important too. :angel:

**Comment by icecrime** [comment]
Alright, everybody gets a warning ;-)

- In `hack/make.sh` when the variable is detected

```
root@116aca1a33c7:/go/src/github.com/docker/docker# DOCKER_EXPERIMENTAL=1 hack/make.sh binary

# WARNING! DOCKER_EXPERIMENTAL is set: building experimental features

bundles/1.7.0-dev already exists. Removing.
```

- At docker start

```
root@116aca1a33c7:/go/src/github.com/docker/docker# docker -d
WARN[0000] Running experimental build
```

- In `docker info`

```
root@116aca1a33c7:/go/src/github.com/docker/docker# docker info
Containers: 0
Images: 0
[...]
Experimental: true
```

**Comment by tianon** [comment]
LGTM :heart:

**Comment by thaJeztah** [comment]
image

**Pull request merged**